ansible_base/lib/redis/client.py

```diff
@@ -0,0 +1,105 @@
+import copy
+import logging
+from typing import Union
+from urllib.parse import urlparse
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.translation import gettext as _
+from django_redis.client import DefaultClient
+from redis import Redis
+from redis.cluster import ClusterNode, RedisCluster
+from redis.exceptions import RedisClusterException
+
+logger = logging.getLogger('ansible_base.lib.redis.client')
+
+
+# We are going to build our own cluster class to override the mget function
+# In a redis cluster, keys might not be in the same slot and this will throw off mget.
+# Instead, we are going to try and use mget and then, if we get the slot error, we will try the mget_nonatomic to make it work
+class DABRedisCluster(RedisCluster):
+    def mget(self, *args, **kwargs):
+        try:
+            return super().mget(*args, **kwargs)
+        except RedisClusterException as e:
+            if 'MGET - all keys must map to the same key slot' in str(e):
+                return super().mget_nonatomic(*args, **kwargs)
+            raise e
```

Comment:
You don't need the argument here, just:
```suggestion
            raise
```
works here.